runtime/src/append_vec.rs

```diff
@@ -185,9 +163,50 @@ impl AppendVec {
             append_offset: Mutex::new(current_len),
             current_len: AtomicUsize::new(current_len),
             file_size: 0, // will be filled by set_file()
+            read_only,
         }
     }
 
+    fn new_empty_map_rw(current_len: usize) -> Self {
+        Self::new_empty_map(current_len, false)
+    }
+
+    pub fn new_empty_map_ro(current_len: usize) -> Self {
+        Self::new_empty_map(current_len, true)
+    }
+
+    pub fn open_and_grow(path: &Path, should_recreate: bool, size: usize) -> File {
+        if should_recreate {
+            let _ignored = remove_file(path);
+        }
+
+        let mut data = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(should_recreate)
+            .open(path)
+            .map_err(|e| {
+                let mut msg = format!("in current dir {:?}\n", std::env::current_dir());
+                for ancestor in path.ancestors() {
+                    msg.push_str(&format!("{:?} is {:?}\n", ancestor, metadata(ancestor)));
+                }
+                panic!(
+                    "{}Unable to {} data path {}, err {:?}",
+                    msg,
+                    if should_recreate { "create" } else { "open" },
+                    path.display(),
+                    e
+                );
+            })
+            .unwrap();
+
+        data.seek(SeekFrom::Start((size - 1) as u64)).unwrap();
```

Comment:
@sakridge I'm aware this should be `fs::truncate` or linux's `fallocate` but this is separate issue... :p